edx_oauth_client/backends/edx_oauth_client.py

```diff
@@ -21,117 +30,153 @@
-        return super(EdxOAuthBackend, self).auth_complete(*args, **kwargs)
+        params, headers = None, None
 
-    def user_data(self, access_token, *args, **kwargs):
-        """ Grab user profile information from SSO. """
-        data = self.get_json(
-            '{}{}'.format(self.OAUTH_PROVIDER_URL, settings.FEATURES.get('OAUTH_USER_DATA_URL', '/login')),
-            params={'access_token': access_token}, method='POST'
+        if self.setting("USER_DATA_REQUEST_METHOD", "GET") == "GET":
+            headers = {'Authorization': 'Bearer {}'.format(access_token)}
+        else:
+            params = {'access_token': access_token}
+
+        data = self.request_access_token(
+            self.setting('USER_DATA_URL'),
+            params=params,
+            headers=headers,
+            method=self.setting("USER_DATA_REQUEST_METHOD", "GET")
         )
+
+        if isinstance(data, list):
+            data = data[0]
+
         if data.get('success') and 'user' in data:
             data = data['user']
         elif 'data' in data:
             data = data['data']
+
         data['access_token'] = access_token
         data.pop('password', None)
+
         return data
 
     def pipeline(self, pipeline, pipeline_index=0, *args, **kwargs):
         self.strategy.session.setdefault('auth_entry', 'register')
-        return super(EdxOAuthBackend, self).pipeline(
+        return super(GenericOAuthBackend, self).pipeline(
             pipeline=self.PIPELINE, *args, **kwargs
         )
 
     def get_user_id(self, details, response):
-        """Return a unique ID for the current user, by default from server
-        response."""
+        """
+        Return a unique ID for the current user, by default from server response.
+        """
         if 'data' in response:
-            return response['data'][0].get(self.ID_KEY)
-        else:
-            return response.get(self.ID_KEY)
+            return response['data'][0].get(self.setting("ID_KEY"))
+
+        return response.get(self.setting("ID_KEY"))
 
     def auth_complete_params(self, state=None):
-        client_id, client_secret = self.get_key_and_secret()
-        return {
-            'grant_type': 'authorization_code',  # request auth code
-            'code': self.data.get('code', ''),  # server response code
-            'id': client_id,
-            'secret': client_secret,
-            'redirect_uri': self.get_redirect_uri(state)
-        }
+        """
+        Update auth complete params from custom oauth provider needs.
+        """
+        res = super(GenericOAuthBackend, self).auth_complete_params(state)
+
+        res.update({
+            'id': res.get('client_id'),
+            'secret': res.get('client_secret'),
+        })
+
+        return res
 
     @handle_http_errors
     def auth_complete(self, *args, **kwargs):
-        """Completes loging process, must return user instance"""
+        """
+        Completes login process, must return user instance
```

Comment:
So and here. Sometimes u add `+s` to a verb and sometimes no.
`Completes`, `Update`. If we are talking about a function so `Completes` is good, `Completes` is bad.

Reply:
I meant `Complete` is bad english.